skbio/metadata/_interval.py

```diff
@@ -0,0 +1,756 @@
+        '''
+        if self.dropped:
+            s = ('{}(dropped=True, locations={!r}, '
+                 'boundaries={!r}, metadata={!r})')
+            return s.format(self.__class__.__name__,
+                            self.locations, self.boundaries, self.metadata)
+        else:
+            s = ('{}(interval_metadata=<{!r}>, locations={!r}, '
+                 'boundaries={!r}, metadata={!r})')
+            return s.format(self.__class__.__name__,
+                            id(self._interval_metadata),
+                            self.locations, self.boundaries, self.metadata)
+
+    @experimental(as_of='0.5.0-dev')
+    def drop(self):
+        '''Drop this ``Interval`` object from the interval metadata it links to.
+
+        If the ``Interval`` object is dropped, you can still get values of
+        ``locations``, ``boundaries``, and ``metadata`` attributes, but you
+        can not change their values with the setters.
+
+        See Also
+        --------
+        skbio.metadata.IntervalMetadata.drop
+        '''
+        if not self.dropped:
+            self._interval_metadata.drop([self])
+
+    def _locations_boundaries_setter(self, locations=None, boundaries=None):
+        if self.dropped:
+            raise RuntimeError('Cannot change `locations` or `boundaries` '
+                               'on a dropped Interval object.')
+        # Casts to `list`, validation, sorting, and setting of `locations`
+        # and `boundaries` happen here.
+        if locations is not None:
+            # check iterability
+            try:
+                # check iterability
+                locations = list(locations)
+            except TypeError:
+                raise TypeError('Cannot give an non-iterable (%r) '
+                                'to `locations`.' % locations)
+
+            # check it is not empty
+            if not locations:
+                raise ValueError('Cannot give empty `locations`.')
+            # check each contiguous span is in right format
+            for location in locations:
+                _assert_valid_location(location)
+
+            spans = len(locations)
+        else:
+            spans = len(self.locations)
+
+        if boundaries is not None:
+            try:
+                boundaries = list(boundaries)
+            except TypeError:
+                raise TypeError(
+                    'Cannot give a non-iterable (%r) '
+                    'to `boundaries`.' % boundaries)
+
+            if len(boundaries) != spans:
+                raise ValueError(
+                    'The length of boundaries must '
+                    'be equal to the length of locations.')
+
+            for boundary in boundaries:
+                _assert_valid_boundary(boundary)
+
+        if locations is None:
+            # `locations` and `boundaries` cannot both be omitted.
+            if boundaries is None:
+                raise ValueError('You must give `None` to both `locations` '
```

Comment:
This error message is inverted, it should say *You cannot provide `None` to both `locations` and `boundaries`*. It's very minor and unlikely to be encountered by users. I don't want this to hold up merge as this is my only remaining comment. It'd be great if you fixed it in the next PR (mixin).